allocator.go

```diff
@@ -78,37 +78,102 @@ type Allocator struct {
 	opts options
 }
 
-// NewRange builds and returns ranges from the necessary parameters.
-func NewRange(id RangeID, rf int, tags []string, demands map[Resource]int64) Range {
-	return Range{
+type NodeMap map[NodeID]Node
+type RangeMap map[RangeID]Range
+
+type Cluster interface{
+	AddNode()
+	UpdateNodeTags () bool
+	UpdateNodeResources() bool
+	RemoveNode () bool
+	AddRange ()
+	RemoveRange () bool
+}
+
+// AddNode Add a new Node
+func (nmap NodeMap) AddNode(id NodeID, tags []string, resources map[Resource]int64) {
+	nmap[id] = Node {
+		id:        id,
+		tags:      tags,
+		resources: resources,
+	}
+}
+
+// UpdateNodeTags Update tags of a Node
+func (nmap NodeMap) UpdateNodeTags(id NodeID, tags []string) bool {
+	if _, found := nmap[id]; found {
+		fmt.Println("Updating Node Tag ", id)
+		nmap[id]= Node {
+			id:        id,
+			tags:      tags,
+			resources: nmap[id].resources,
+		}
```

Comment:
it was giving an error

Reply:
you cant do that

Reply:
Oh right, because the type is a `map`, I'd suggest changing to a mapping from `nodeID/rangeID` to a pointer onto `Node/Range`, that should work, I just tested on my local.

Context : https://stackoverflow.com/questions/32751537/why-do-i-get-a-cannot-assign-error-when-setting-value-to-a-struct-as-a-value-i